FWCore/Framework/src/StreamSchedule.cc

```diff
@@ -207,16 +205,11 @@ namespace edm {
       // 2) if it is a WorkerT<EDProducer>, add it to our list
       // 3) hand list to our delayed reader
       for (auto const& label : unusedLabels) {
```

Comment:
Actually it is good because at this point it still includes EDAnalyzers and OutputModules in addition to EDProducers and EDFilters. Inside workerManager_.addToUnscheduledWorkers it splits the list into unscheduledLabels and shouldBeUsedLabels.